.pre-commit-config.yaml

```diff
@@ -0,0 +1,40 @@
+---
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.4.0
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: mixed-line-ending
+      - id: check-byte-order-marker
+      - id: check-executables-have-shebangs
+      - id: check-merge-conflict
+      - id: check-symlinks
+      - id: debug-statements
+      - id: check-yaml
+        files: .*\.(yaml|yml)$
+        args: [--allow-multiple-documents]
+  - repo: https://github.com/pycqa/flake8
+    rev: 3.9.0
+    hooks:
+      - id: flake8
+        additional_dependencies: [flake8-typing-imports==1.6.0]
+        # List of ignored checks overrides the flake8 defaults.
+        # Changes to the list, especially extensions should be
+        # justified with relation to defaults.
+        # E501 - line is too long, would require local ignore or obfuscating changes
+        # W503 - line break before binary operator, ignored by default
+        entry: flake8 --ignore=E501,W503
+  - repo: https://github.com/openstack-dev/bashate
+    rev: 2.1.1
+    hooks:
+      - id: bashate
+        entry: bashate --error . --ignore=E006,E040
```

Comment:
environment.d/*.bash files are sourced, not executed. So I think E005 should not be enforced for these files. It looks like the only option is to add E005 to --ignore

Reply:
Done. I will squash the commits after I get approval.